pkg/validations/base_test.go

```diff
@@ -85,13 +87,36 @@ func createTestDeployment(replicas int32) (*appsv1.Deployment, error) {
 	return &d, nil
 }
 
+func intializeEngine(t *testing.T, customCheck ...config.Check) {
+
+	// Reset global prometheus registry to avoid testing conflicts
+	metrics.Registry = prometheus.NewRegistry()
+
+	// Check if custom check has been set
+	if len(customCheck) > 0 {
+		// Initialize engine with custom check
+		e, err := newEngine(newEngineConfigWithCustomCheck(customCheck[0]))
```

Comment:
Would actually be better to refactor this to support multiple custom checks vs just always using the first one here.  Doing this makes the function signature misleading.

Reply:
@rrati Can you double check the new changes? I put a loop in to iterate through the rest of the list once the engine is created if there are more custom checks.

Reply:
There's no need to do this.  The custom checks in the config is an array, and what is passed in to this function is an array.  I would just modify newEngineWithCustomCheck to accept an array of check configs.

Reply:
@rrati Changed it so that it fills in the newEngineWithCustomCheck function. Can you check it and see if that's what you were looking for?

Reply:
This is still only using 1 check.  I was looking more for something like:

newEngineConfigWithCustomCheck(customCheck []config.Check) {
...
customConfig := config.Config{
		CustomChecks: customCheck,
...
...

Reply:
@rrati Ah yes, I was over complicating it. The latest change should be using the array properly now.